jenkins/run

```diff
@@ -21,16 +16,28 @@ if [[ -z "${IMAGE_REPO}" ]]; then
     IMAGE_REPO="quay.io/pbench"
 fi
 
-# We need a unique working directory for this run.
-mkdir -p ${WORKSPACE_TMP}
-_run_dir=$(mktemp -d -p ${WORKSPACE_TMP})
+if [[ -z "${BUILD_NUMBER}" ]]; then
+    # Not in a Jenkins job ...
+    WORKSPACE_TMP=/tmp/${USER}/jenkins-run
+    BUILD_NUMBER=0
+else
+    if [[ -z "${WORKSPACE_TMP}" ]]; then
+        printf -- "ERROR - missing required Jenkins environment variable, WORKSPACE_TMP\n" >&2
+        exit 1
+    fi
+fi
+
+# Ensure the given workspace temporary directory exists.
+_run_dir="${WORKSPACE_TMP}/${BUILD_NUMBER}"
+mkdir -p ${_run_dir}
+
+# We need a unique working directory for this run, provided by the BUILD_NUMBER.
```

Comment:
Of course if I were silly enough to run the tests manually twice simultaneously, they both have `/tmp/dbutenho/jenkins-run/0` ... but "close enough for government work". (Or horseshoes and hand grenades...)

Reply:
Yes, don't do that.  =)

We could consider see if that numbered directory is there, and auto-increment.  But running the jenkins modes is not really common.